### DISCUSSION TITLE
unable to bypass negatives

### DISCUSSION THREAD

**Discussion by shi-yan** [pest-parser/pest#627] [General]
I have the following grammar
```
sequence_body = {sequence_begin ~ NEWLINE+ ~ (!sequence_end ~ sequence_line ~ NEWLINE+)* ~ sequence_end}
```
I expect it to match the following:

```
sequence_begin
sequence_line
sequence_line
sequence_line
sequence_line
...
sequence_end
```

because sequence_line starts with ASCII_ALPHA+, and sequence_end starts with "end", which is also ASCII_ALPHA+, I added !sequence_end

And I got the error:

unexpected sequence_end

it refers to the actual sequence_end

```
Error { variant: ParsingError { positives: [], negatives: [sequence_group_end] }, location: Pos(317), line_col: Pos((15, 9)), path: None, line: "        end␊", continued_line: None }'
```
`(!sequence_end ~ line) * ~ sequence_end`

if the parser hits the first !sequence_end, I would expect it to finish the `()*` part and then match the second positive sequence_end.

but instead, it just generates an error.

[plantumlparser.zip](https://github.com/pest-parser/pest/files/6168387/plantumlparser.zip)

the reproducible project is attached too.

**Comment by 01mf02** [comment]
Hello,

I have tried to reproduce your problem.

I made the following grammar:

~~~
sequence_body = {sequence_begin ~ NEWLINE+ ~ (!sequence_end ~ sequence_line ~ NEWLINE+)* ~ sequence_end}
sequence_begin = {"begin"}
sequence_line = {"line"}
sequence_end = {"end"}
~~~

And the following program:

~~~ rust
extern crate pest;
#[macro_use]
extern crate pest_derive;

use pest::Parser;

#[derive(Parser)]
#[grammar = "grammar.pest"]
pub struct SeqParser;

const SEQ: &str =
"begin
line
line
end
";

fn main() {
    let parse = SeqParser::parse(Rule::sequence_body, SEQ);
    match parse {
        Err(e) => println!("error: {}", e),
        Ok(_) => println!("done"),
    }
}
~~~

However, this seems to work fine.
So perhaps it might help you to adapt my example until you experience your error?
